src/util/virsocket.c

```diff
@@ -485,6 +488,94 @@ virSocketRecvFD(int sock, int fdflags)
 
     return fd;
 }
+
+/* virSocketSendMsgWithFD send FDs along with payload on socket.
+   Return number of bytes sent on success, or -1 with errno set in case of error.
+*/
+int
+virSocketSendMsgWithFD(int sock, const char *payload, int *fds, size_t fd_len)
+{
+    struct msghdr msg;
+    struct iovec iov[1];
+    int ret;
+    char control[CMSG_SPACE(sizeof(int)*fd_len)];
+    struct cmsghdr *cmsg;
+
+    memset(&msg, 0, sizeof(msg));
+    memset(control, 0, sizeof(control));
+
+    iov[0].iov_base = (void *)payload;
+    iov[0].iov_len = strlen(payload);
+
+    msg.msg_iov = iov;
+    msg.msg_iovlen = 1;
+
+    msg.msg_control = control;
+    msg.msg_controllen = sizeof(control);
+
+    cmsg = CMSG_FIRSTHDR(&msg);
+    if (!cmsg) {
+        perror("CMSG_FIRSTHDR returned NULL");
+        return -1;
+    }
+    cmsg->cmsg_len = CMSG_LEN(sizeof(int)*fd_len);
+    cmsg->cmsg_level = SOL_SOCKET;
+    cmsg->cmsg_type = SCM_RIGHTS;
+    memcpy(CMSG_DATA(cmsg), fds, sizeof(int) * fd_len);
+
+    do {
+        ret = sendmsg(sock, &msg, 0);
+    } while (ret < 0 && errno == EINTR);
+
+    if (ret < 0)
+   {
+      perror("sendmsg() failed");
+   }
+
+    return ret;
+}
+
+/* Return HTTP response from recieved message, or -1 on Error
+*/
+int virSocketRecvHttpResponse(int sock) {
+    struct pollfd pfds[1];
```

Comment:
A comment would be nice why we are using array size 1

Reply:
Since this method monitors and receives messages from a single socket, I think it is rather straightforward from code.

I will leave it as is, unless you are strongly suggesting this.

Reply:
It's okay as is.